float64 epsilon
---
bool ret_value